flight/PiOS/Common/pios_openlrs.c

```diff
@@ -526,11 +528,17 @@ static bool binding_button_pushed(pios_openlrs_t openlrs_dev)
 			openlrs_dev->cfg.bind_button->init.GPIO_Pin) !=
 		Bit_RESET;
 
-	if (openlrs_dev->cfg.bind_active_high) {
-		return truth_value;
-	} else {
-		return !truth_value;
+	if (!openlrs_dev->cfg.bind_active_high) {
+		truth_value = !truth_value;
 	}
+
+	button_history[2] = button_history[1];
+	button_history[1] = button_history[0];
+	button_history[0] = truth_value;
+
+	/* Catch an edge, but also require 2 in a row. */
```

Comment:
AKA debounce. The usual method of counting cycles seems cleaner than an array IMO, especially if you want more cycles... which IMO we should require something like 100 ms.

```c
static int button_debounce = 0;
if (truth_value)
	button_debounce++;
else
	button_debounce = 0;

return button_debounce >= 3;
```

Reply:
Yah, this is pretty reliable in my test so far.  I can change variants if you really care.

Reply:
No, not really. It just seemed like a really weird way to do it. :stuck_out_tongue:

Reply:
:D it looks like when I write HDL...